game-core/src/test/java/games/strategy/triplea/delegate/power/calculator/TotalPowerAndTotalRollsTest.java

```diff
@@ -1,33 +1,690 @@
+          dice, sortedDie, false, unitPowerAndRollsMap, validTargets, gameData, true);
+    }
+
+    @Test
+    void singleAaWithOneRollNoHit() {
+      final GameData gameData = givenGameData().withDiceSides(6).build();
+      final Unit unit = givenUnit("test", gameData);
+      unit.getUnitAttachment().setOffensiveAttackAa(2).setMaxAaAttacks(1);
+      final List<Unit> units = List.of(unit);
+      final List<Die> sortedDie = new ArrayList<>();
+
+      final Triple<Integer, Integer, Boolean> result =
+          whenGetPowerHitsResult(gameData, units, sortedDie, 6, 4);
+
+      assertThat(result, is(Triple.of(2, 0, true)));
+      assertThat(sortedDie, is(List.of(new Die(6, 2, Die.DieType.MISS))));
+    }
+
+    @Test
+    void singleAaWithTwoRoll() {
+      final GameData gameData = givenGameData().withDiceSides(6).build();
+      final Unit unit = givenUnit("test", gameData);
+      unit.getUnitAttachment().setOffensiveAttackAa(2).setMaxAaAttacks(2);
+      final List<Unit> units = List.of(unit);
+      final List<Die> sortedDie = new ArrayList<>();
+
+      final Triple<Integer, Integer, Boolean> result =
+          whenGetPowerHitsResult(gameData, units, sortedDie, 1, 4);
+
+      assertThat(result, is(Triple.of(4, 2, true)));
+      assertThat(
+          sortedDie, is(List.of(new Die(1, 2, Die.DieType.HIT), new Die(1, 2, Die.DieType.HIT))));
+    }
+
+    @Test
+    void twoAaWithSamePower() {
+      final GameData gameData = givenGameData().withDiceSides(6).build();
+      final Unit unit = givenUnit("test", gameData);
+      unit.getUnitAttachment().setOffensiveAttackAa(2).setMaxAaAttacks(1);
+      final Unit unit2 = givenUnit("test2", gameData);
+      unit2.getUnitAttachment().setOffensiveAttackAa(2).setMaxAaAttacks(1);
+      final List<Unit> units = List.of(unit, unit2);
+      final List<Die> sortedDie = new ArrayList<>();
+
+      final Triple<Integer, Integer, Boolean> result =
+          whenGetPowerHitsResult(gameData, units, sortedDie, 1, 4);
+
+      assertThat(result, is(Triple.of(4, 2, true)));
+      assertThat(
+          sortedDie, is(List.of(new Die(1, 2, Die.DieType.HIT), new Die(1, 2, Die.DieType.HIT))));
+    }
+
+    @Test
+    void twoAaWithDifferentPower() {
+      final GameData gameData = givenGameData().withDiceSides(6).build();
+      final Unit unit = givenUnit("test", gameData);
+      unit.getUnitAttachment().setOffensiveAttackAa(2).setMaxAaAttacks(1);
+      final Unit unit2 = givenUnit("test2", gameData);
+      unit2.getUnitAttachment().setOffensiveAttackAa(3).setMaxAaAttacks(1);
+
+      final List<Unit> units = List.of(unit, unit2);
+      final List<Die> sortedDie = new ArrayList<>();
+
+      final Triple<Integer, Integer, Boolean> result =
+          whenGetPowerHitsResult(gameData, units, sortedDie, 1, 4);
+
+      assertThat(result, is(Triple.of(5, 2, false)));
+      assertThat(
+          sortedDie, is(List.of(new Die(1, 3, Die.DieType.HIT), new Die(1, 2, Die.DieType.HIT))));
```

Comment:
FYI there is a matcher to check list elements, I believe it is `IsIterableContaining`